internal/handlers/common/update.go

```diff
@@ -16,36 +16,48 @@ package common
 
 import (
 	"fmt"
+	"sort"
 	"strings"
 
 	"github.com/FerretDB/FerretDB/internal/types"
-	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
 // UpdateDocument updates the given document with a series of update operators.
-func UpdateDocument(doc, update *types.Document) error {
+// Returns true if document was changed.
+func UpdateDocument(doc, update *types.Document) (bool, error) {
 	for _, updateOp := range update.Keys() {
 		updateV := must.NotFail(update.Get(updateOp))
 
 		switch updateOp {
 		case "$set":
-			setDoc, err := AssertType[*types.Document](updateV)
-			if err != nil {
-				return err
-			}
 
-			for _, setKey := range setDoc.Keys() {
-				setValue := must.NotFail(setDoc.Get(setKey))
-				if err = doc.Set(setKey, setValue); err != nil {
-					return lazyerrors.Error(err)
+			switch setDoc := updateV.(type) {
+			case *types.Document:
+				if setDoc.Len() == 0 {
+					return false, nil
+				}
+				var err error
```

Comment:
`err` variable is not used anywhere outside the `for` block. We can remove it.

Reply:
done.